### PR TITLE
Partial scene date

### PR DESCRIPTION
**Pull request by peolic**
~~As stated on Discord, still need to verify drafts work.~~
Draft->Edit works as intended.

### CODE REVIEW

File: frontend/src/pages/scenes/sceneForm/SceneForm.tsx [hunk 1]
```diff
@@ -273,6 +278,10 @@ const SceneForm: FC<SceneProps> = ({ scene, initial, callback, saving }) => {
               <Form.Control.Feedback type="invalid">
                 {errors?.date?.message}
               </Form.Control.Feedback>
+              {/* <Form.Text>
+                If the precise date is unknown the day and/or month can be
+                omitted.
+              </Form.Text> */}
```
Comment (peolic) [inline frontend/src/pages/scenes/sceneForm/SceneForm.tsx:297]: Copied this from `PerformerForm`, but there isn't enough space for this text.
The error message for invalid date is kind of too long now, as well.

File: pkg/models/model_scene.go [hunk 2]
```diff
@@ -214,8 +215,26 @@ func CreateScenePerformers(sceneID uuid.UUID, appearances []*PerformerAppearance
...
-	p.Date = SQLiteDate{String: date, Valid: true}
+func (p *Scene) setDate(fuzzyDate FuzzyDateInput) {
+	p.Date = SQLiteDate{String: fuzzyDate.Date, Valid: fuzzyDate.Date != ""}
+	p.DateAccuracy = sql.NullString{String: fuzzyDate.Accuracy.String(), Valid: fuzzyDate.Date != ""}
+}
+
+func (p Scene) ResolveDate() *FuzzyDate {
+	if !p.Date.Valid {
+		return nil
+	}
```
Comment (peolic) [inline pkg/models/model_scene.go:226]: This change might affect Stash, if Stash expects date to never be null.
However, the schema for Stash-Box already has the date field defined as nullable.

**Comment by peolic** [comment]
Stash-Box end works.
This, however, will require an update to Stash because date is no longer a String.
https://github.com/stashapp/stash/blob/c62b0b28ed27d8d37b252dd684998db7c6e28ee1/graphql/stash-box/query.graphql#L94-L99
```gql
date
```
vs
```gql
date {
  date
  accuracy
  # or
  ...FuzzyDateFragment
}
```

I don't particularly like this, but I could keep `date` a String and add a `date_accuracy`, which _is_ backwards compatible.
I would rather have `date` be similar to the performer birthdate schema, I would have to see if it's possible to have a union of a scalar (String) with an interface (FuzzyDate) and return a different type based on the request.

**Comment by peolic** [comment]
I added `DateFuzzy` and deprecated `Date` instead. This is not very elegant, but it's backwards compatible.

**Review by ghost — APPROVED** [review]

**Pull request merged**